sklearnex/cluster/k_means.py

```diff
@@ -110,13 +110,13 @@ def _onedal_fit_supported(self, method_name, X, y=None, sample_weight=None):
             ) or not issparse(X)
 
             _acceptable_sample_weights = True
-            if sample_weight is not None or not isinstance(sample_weight, numbers.Number):
+            if (sample_weight is not None) and (
+                not isinstance(sample_weight, numbers.Number)
+            ):
                 sample_weight = _check_sample_weight(
                     sample_weight, X, dtype=X.dtype if hasattr(X, "dtype") else None
                 )
-                _acceptable_sample_weights = np.allclose(
-                    sample_weight, np.ones_like(sample_weight)
-                )
+                _acceptable_sample_weights = np.all(sample_weight == 1)
```

Comment:
This is not applicable for constant values that differ from 1?
```suggestion
                _acceptable_sample_weights = np.all(sample_weight == sample_weight[0])
```